app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.java

```diff
@@ -1811,25 +1813,28 @@ public void onVideoTrackDisabled(
         }
     }
 
-    private @DrawableRes int getNetworkQuality(NetworkQualityLevel networkQualityLevel) {
-        int result = R.drawable.network_quality_level_0;
-
+    private void setNetworkQualityLevelImage(
+            ImageView networkQualityImage, NetworkQualityLevel networkQualityLevel, String sid) {
+        networkQualityLevels.put(sid, networkQualityLevel);
         if (networkQualityLevel == NetworkQualityLevel.NETWORK_QUALITY_LEVEL_UNKNOWN
                 || networkQualityLevel == NetworkQualityLevel.NETWORK_QUALITY_LEVEL_ZERO) {
-            result = R.drawable.network_quality_level_0;
+            networkQualityImage.setVisibility(View.GONE);
```

Comment:
If NQL drops to zero, you want to show the 0 image. This is important for remote participants, if somebody drops off the network we should show the others `R.drawable.network_quality_level_0`

I would split these two cases:

```
if (networkQualityLevel == NetworkQualityLevel.NETWORK_QUALITY_LEVEL_UNKNOWN) {
    networkQualityImage.setVisibility(View.GONE);
} else if (networkQualityLevel == NetworkQualityLevel.NETWORK_QUALITY_LEVEL_ZERO) {
    networkQualityImage.setVisibility(View.VISIBLE);
    networkQualityImage.setImageResource(R.drawable.network_quality_level_0);
} else if.....
```